File: app/src/main/java/com/example/crashhandler/MainActivity.java
package com.example.crashhandler;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button btnCrash;
    Context ctxMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ctxMain = this;
        Thread.setDefaultUncaughtExceptionHandler(new CrashExceptionHandler(this));

        btnCrash = findViewById(R.id.btnCrash);

        btnCrash.setOnClickListener(v -> {
            String s = null;
            Log.d("ThisLogWillCrashTheApp" , "" + s.length());
        });

    }
}

File: app/src/main/java/com/example/crashhandler/MainActivity.kt
package com.example.crashhandler

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import com.example.crashhandler.databinding.ActivityMainBinding
import java.lang.NullPointerException

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        val context: Context = this

        Thread.setDefaultUncaughtExceptionHandler(CrashHandler(context))

        val timer = object : CountDownTimer(5000, 1000) {
            override fun onTick(p0: Long) {
                binding.txtCountDown.text = resources.getString(R.string.count_down, p0/1000)
            }
            override fun onFinish() {
                throw NullPointerException()
            }
        }
        timer.start()

    }
}